bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceRegistryOSGITest.groovy

```diff
@@ -440,5 +440,55 @@ class DiscoveryServiceRegistryOSGITest extends OSGiTest {
         assertNotNull extendedDiscoveryServiceMock.discoveryServiceCallback.getExistingDiscoveryResult(thingUID)
     }
 
+    @Test
+    void 'assert that updated discovery result is stored properly in cache'() {
+        ThingUID thingUID = new ThingUID(EXTENDED_BINDING_ID, EXTENDED_THING_TYPE, "foo")
+        
+        
+        AsyncResultWrapper<DiscoveryResult> discoveryListenerResult = new AsyncResultWrapper<DiscoveryResult>()
+        DiscoveryListener discoveryListenerMock = [
+            thingDiscovered: { DiscoveryService source, DiscoveryResult result ->
+                discoveryListenerResult.set result
+
+            },
+            removeOlderThings: { DiscoveryService source, long timestamp, Collection<ThingTypeUID> thingTypeUIDs ->
+                []
+            }
+        ] as DiscoveryListener
+
+        boolean discoveryFinished = false;
+        discoveryServiceRegistry.startScan(new ThingTypeUID(EXTENDED_BINDING_ID, EXTENDED_THING_TYPE), [
+            onFinished: { discoveryFinished = true },
+            onErrorOccurred: { }
+        ] as ScanListener);
+        waitForAssert ({ assertTrue(discoveryFinished) }, 2000 );
+        
+        discoveryServiceRegistry.addDiscoveryListener(discoveryListenerMock);
+        waitForAssert ({ assertNotNull(discoveryListenerResult.getWrappedObject()) }, 2000 );
+        
+        assertEquals new HashMap<>(), discoveryListenerResult.getWrappedObject().getProperties();
+        
+        discoveryServiceRegistry.removeDiscoveryListener(discoveryListenerMock);
+       
+        discoveryListenerResult.set null;
+        
+        Map<String, Object> props = new HashMap<>();
+        props.put("ip", "0.0.0.0");
+        
+        extendedDiscoveryServiceMock.setDiscoveryProperties(props);
+        
+        discoveryFinished = false;
+        discoveryServiceRegistry.startScan(new ThingTypeUID(EXTENDED_BINDING_ID, EXTENDED_THING_TYPE), [
```

Comment:
factor the `ScanListener` out, so you can reuse the one from above (easier to read)